component/componenttest/nop_host.go

```diff
@@ -40,3 +40,7 @@ func (nh *nopHost) GetExtensions() map[config.ComponentID]component.Extension {
 func (nh *nopHost) GetExporters() map[config.DataType]map[config.ComponentID]component.Exporter {
 	return nil
 }
+
+func (nh *nopHost) RegisterStatusReporter(reporter component.StatusReportFunc) {}
```

Comment:
The name `StatusReporter` does not sound right to me. This is not a _reporter_ of the status, it is a _consumer_ or _receiver_ of the status, isn't it?